qemu/tests/balloon_sc_interrogate.py

```diff
@@ -0,0 +1,70 @@
+import re
+import logging
+
+from avocado.core import exceptions
+from virttest import utils_test
+from virttest import error_context
+from qemu.tests.balloon_check import BallooningTestWin
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Balloon service should be in running status after sc interrogate it.
+    1) boot a guest with balloon device.
+    2) check balloon driver installation status.
+    3) enable and check driver verifier in guest.
+    4) install and start balloon service in guest.
+    5) send INTERROGATE signal to balloon service.
+    6) check balloon service status again.
+    """
+
+    def interrogate_balloon_service(session):
+        """
+        Sending INTERROGATE to balloon service.
+        :param session: VM session.
+        """
+        logging.info("Send INTERROGATE to balloon service")
+        sc_interrogate_cmd = params["sc_interrogate_cmd"]
+        status, output = session.cmd_status_output(sc_interrogate_cmd)
+        if status:
+            test.error(output)
+
+    error_context.context("Boot guest with balloon device", logging.info)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login()
+    driver_name = params.get("driver_name", "balloon")
+
+    utils_test.qemu.windrv_verify_running(session, test, driver_name)
+    utils_test.qemu.setup_win_driver_verifier(driver_name, vm)
+
+    err = None
+    try:
+        # Install and start balloon service in guest
+        balloon_test = BallooningTestWin(test, params, env)
```

Comment:
It should be moved out from this try section because the service wasn't installed and configured at this point.